moleculemaker/mmli-backend
# Add IP ratelimit middleware to ChemScraper Analyze endpoint

## Problem
We would like to ratelimit the ChemScraper Analyze endpoint, while allowing unrestricted access to the other endpoints

## Approach
* split `/chemscraper/analyze` into its own ingress rule
* apply ip-ratelimit middleware to `/chemscraper/analyze` endpoint (prod + staging)
* point staging values at staging OAuth2 Proxy

## How to Test
1. Navigate to https://chemscraper.frontend.staging.mmli1.ncsa.illinois.edu/configuration
2. ~~Login using the link at the top-right (to avoid hcaptcha)~~ no longer needed
3. Upload a PDF and submit for analysis
4. Click the browser's Back button and attempt to submit the same job again (within ~30s of the first submission)
    * You should see that Traefik rejects this request and returns `HTTP 429: Too Many Requests`
    * You should see that a response header is returned telling the user when they can retry the request again
